crates/apub/src/api/search.rs

```diff
@@ -201,6 +201,12 @@ pub async fn search(
     }
   };
 
+  //filtering out removed and deleted comments from search results
+  let comments = comments
+    .into_iter()
+    .filter(|c| !c.comment.removed && !c.comment.deleted)
```

Comment:
Nitpick: you can DeMorgan this to only use 1 negation.

```suggestion
    .filter(|c| !(c.comment.removed || c.comment.deleted))
```